File: handyman.parent/handyman/src/main/scala/in/handyman/util/ExceptionUtil.scala
package in.handyman.util

import java.io.{PrintWriter, StringWriter}

object ExceptionUtil {
  
  def completeStackTraceex(executionException:Throwable):String={
    val errors = new StringWriter();
    val print = new PrintWriter(errors)
    if(executionException!=null) executionException.printStackTrace(print) else print.print("NONE");
    errors.toString();
    
  }
  
  def completeSQLWarning(sqlWarnings:java.sql.SQLWarning):String={
    val errors = new StringWriter();
    val print = new PrintWriter(errors);
    if(sqlWarnings!=null) sqlWarnings.printStackTrace(print) else print.print("NONE");
    errors.toString();
  }
}

File: src/main/java/in/handyman/raven/core/util/ExceptionUtil.java
package in.handyman.raven.core.util;

import java.io.PrintWriter;
import java.io.StringWriter;
import java.sql.SQLWarning;

public class ExceptionUtil {

    public static String toString(final Throwable ex) {
        final StringWriter sw = new StringWriter();
        ex.printStackTrace(new PrintWriter(sw));
        return sw.toString();
    }

    public static String completeSQLWarning(final SQLWarning sqlWarnings) {
        var errors = new StringWriter();
        var print = new PrintWriter(errors);
        if (sqlWarnings != null) {
            sqlWarnings.printStackTrace(print);
        } else {
            print.print("NONE");
        }
        return errors.toString();
    }

}
